src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,183 @@
+     *
+     * @param fileString String to be converted.
+     * @return Task object.
+     */
+    public static Task fileStringToTask(String fileString) {
+        String[] parts = fileString.split(" \\| ");
+        if (parts.length < 3) {
+            ChatbotException.getError(ChatbotException.ErrorType.TASK_CORRUPT);
+        }
+
+        String type = parts[0].trim();
+        boolean isDone = parts[1].trim().equals("1");
+        String description = parts[2];
+
+        Task task = new Task(description);
+
+        switch (type) {
+        case "T":
+            task = createTask("todo", description);
+            break;
+        case "D":
+            if (parts.length < 4) {
+                ChatbotException.getError(ChatbotException.ErrorType.TODO_CORRUPT);
+            }
+            description += " /by " + parts[3];
+            task = createTask("deadline", description);
+            break;
+        case "E":
+            String timeInfo = parts[3].trim();
+            String[] timeParts = timeInfo.split(" - ");
+            if (timeParts.length < 2) {
+                ChatbotException.getError(ChatbotException.ErrorType.EVENT_CORRUPT);
+            }
+            description += " /from " + timeParts[0].trim() + " /to " + timeParts[1].trim();
+            task = createTask("event", description);
+            break;
+        default:
+            ChatbotException.getError(ChatbotException.ErrorType.UNKNOWN_TASK);
+        }
+
+        task.mark(isDone);
+        return task;
+    }
+
+    /**
+     * Returns appropriate Task type given user command
+     *
+     * @param command     User commands.
+     * @param description Task description to be added.
+     * @return Task object.
+     */
+    public static Task createTask(String command, String description) {
+        if (command.startsWith("todo")) {
+            return new ToDo(description);
+
+        } else if (command.startsWith("deadline")) {
+            String[] parts = description.split("/by", 2);
+            LocalDateTime by = DateParser.parseDateTime(parts[1].trim());
+            return new Deadline(parts[0], by);
+
+        } else {
+            String[] parts = description.split("\\s+/from\\s+|\\s+/to\\s+");
+            LocalDateTime from = DateParser.parseDateTime(parts[1].trim());
+            LocalDateTime to = DateParser.parseDateTime(parts[2].trim());
+            return new Event(parts[0], from, to);
+
+        }
+    }
+
+    /**
+     * Checks if there are any saved tasks from the previous run
+     */
+    public static void ensureFileExists() {
+        try {
+            File file = new File(FILE_PATH);
+            if (!file.exists()) {
+                file.getParentFile().mkdirs(); // Ensure the directory exists
+                file.createNewFile(); // Create the file if it doesn't exist
+            }
+        } catch (IOException e) {
+            System.out.println("An error occurred while ensuring the task file exists.");
+        }
```

Comment:
Consider throwing this as a new `ChatbotException` and catching it in your `Chatbot` class
```
public static void ensureFileExists() throws ChatbotException
...
    } catch (IOException e) {
        throw new ChatbotException("An error occurred while ensuring the task file exists.");
    }
}